src/app/Http/Controllers/Admin/SettingController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Resources\Admin\SettingResource;
+use App\Interfaces\SettingRepositoryInterface;
+use App\Models\Setting;
+use Illuminate\Http\Request;
+
+class SettingController extends ResourceController
+{
+    private SettingRepositoryInterface $settingRepository;
+
+    public function __construct(SettingRepositoryInterface $settingRepository)
+    {
+        $this->settingRepository = $settingRepository;
+    }
+
+    public function index(Request $request)
+    {
+        $this->authorize('viewAny', Setting::class);
+
+        $settings = $this->settingRepository->getAll();
+        return response()->json(SettingResource::collection($settings));
+    }
+
+    public function update(Request $request)
+    {
+        $this->authorize('update', Setting::class);
+
+        $validated = validator($request->all(), [
+            '*' => function ($attribute, $value, $fail) {
+                if (!array_key_exists($attribute, Setting::SETTING_TYPES)) {
+                    $fail("The setting key '{$attribute}' is not valid.");
+                }
+            },
+        ])->validate();
+
+        $rules = [];
+        foreach ($validated as $key => $value) {
```

Comment:
You're running a validator against the payload twice. Once to validate the keys, and again to validate the types. You should be able to just do it all in one pass.